### ISSUE TITLE
disable singleplayer tab when unitsync isn't found / selected

### ISSUE THREAD

**Issue by abma** [springlobby/springlobby#469]
it seems to lead to confusion.

not sure if its better to completely disable the tab or print an error message and/or show edit->preferences->spring.

**Comment by Usaga** [comment]
Maybe it is better to print out error message on canvas where map should be drawn and disable all controls on tab?

**Comment by Usaga** [comment]
At least user will be properly informed about something is wrong

**Comment by jamerlan** [comment]
You can show message box for example that no engine was selected/added and give a hint how to do that. Just when user press "start" button for example

**Comment by abma** [comment]
without unitsync no map/game can be selected as well, so imo it should a message before letting the user select game / map.

so the error message on canvas is imo good as popups are pretty annoying / useless.

**Comment by OursDesCavernes** [comment]
I'm trying to get some get do that:
- message with hint to pref when no unitsync in minimap area
- start button disabled
I'll let you know

**Comment by Usaga** [comment]
Some hint of possible variant:
- create method CheckEngineAvailabe or similar in class SinglePLayerTab
- run it from SinglePlayerTab constructor
- in the same constructor subscribe class on UnitSyncReloaded global event. In it's handler call CheckEngineAvailable
- in created method check engine, maps and mod and set state variable, enable\disable start button
- in paint event handler of minimap preview check for state variable and print neede hints

**Comment by OursDesCavernes** [comment]
I'm doing this:
 - add a new panel to the minimap spacer and hide it
 - on engine list refresh, if no engine found, hide the minimap and show the new.
This is working fine, I need to add content to the new panel.
Next step: disable start button.

**Comment by Usaga** [comment]
Sounds good

**Comment by OursDesCavernes** [comment]
Maybe someone with more skill at wx could do some cosmetics ?

**Comment by Usaga** [comment]
WHat do you mean?

**Comment by OursDesCavernes** [comment]
the new panel is an empty window with a basic style and that doesn't fill the available space (just a static 100x100 window).
There is also no explanation, just a short tool-tip.
I tried to get it nicer without success, so I pushed it as is.
It is functional but not nice looking nor user friendly.
That what I meant by cosmetics.
(remember that I'm new to wxwidget, I spend most of the time reading docs and code samples)
